scripts/tests/core-integration.test.mjs

```diff
@@ -0,0 +1,447 @@
+      const testFiles = [
+        "010_user-guide/getting-started.md",
+        "010_user-guide/README.md",
+        "020_actions/mutate.md",
+        "README.md",
+        "index.js",
+      ];
+
+      for (const file of testFiles) {
+        const filePath = path.join(docsDir, file);
+        await fs.mkdir(path.dirname(filePath), { recursive: true });
+        await fs.writeFile(filePath, "# Test\n\nContent", "utf8");
+      }
+
+      const foundFiles = await glob(`${docsDir}/**/*.md`);
+      const relativePaths = foundFiles.map(f => path.relative(docsDir, f));
+
+      expect(relativePaths).toContain("010_user-guide/getting-started.md");
+      expect(relativePaths).toContain("010_user-guide/README.md");
+      expect(relativePaths).toContain("020_actions/mutate.md");
+    });
+
+    it("should exclude non-.md files", async () => {
+      const foundFiles = await glob(`${docsDir}/**/*.md`);
+      const foundBasenames = foundFiles.map(f => path.basename(f));
+
+      expect(foundBasenames).not.toContain("index.js");
+    });
+
+    it("should filter out root README.md from processing", async () => {
+      const files = ["README.md", "010_user-guide/README.md", "getting-started.md"];
+
+      // Simulate the filtering logic from findSourceDocFiles
+      const filtered = files.filter(f => f.endsWith(".md")).filter(f => !(f === "README.md"));
+
+      expect(filtered).not.toContain("README.md");
+      expect(filtered).toContain("010_user-guide/README.md");
+      expect(filtered).toContain("getting-started.md");
+    });
+  });
+
+  describe("Root Community Files", () => {
+    it.each([
+      ["SECURITY.md", "# Security Policy\n\nPlease report vulnerabilities...", "Security Policy"],
+      ["CODE-OF-CONDUCT.md", "# Code of Conduct\n\nBe respectful...", "Code of Conduct"],
+      ["SUPPORT.md", "# Support\n\nGet help with Pepr...", "Support"],
+    ])("should process %s from repository root", async (filename, content, expectedText) => {
+      const filePath = path.join(coreRepo, filename);
+
+      await fs.writeFile(filePath, content, "utf8");
+
+      const exists = await fs
+        .access(filePath)
+        .then(() => true)
+        .catch(() => false);
+      expect(exists).toBe(true);
+
+      const fileContent = await fs.readFile(filePath, "utf8");
+      expect(fileContent).toContain(expectedText);
+    });
+  });
+
+  describe("File Processing with Metadata", () => {
+    it("should strip numbered prefixes from directories", async () => {
+      const sourceFile = path.join(docsDir, "010_user-guide/getting-started.md");
+      await fs.mkdir(path.dirname(sourceFile), { recursive: true });
+      await fs.writeFile(sourceFile, "# Getting Started\n\nContent...", "utf8");
+
+      // Simulate metadata generation
+      const relativePath = path.relative(docsDir, sourceFile);
+      const cleanPath = relativePath.replace(/\d+_/g, "");
+
+      expect(cleanPath).toBe("user-guide/getting-started.md");
+    });
```

Comment:
In comparison to the other tests that check implementation details, I think this test is okay since we're providing a regular-expression that we define. Executable code to show that a regex behaves as expected reduces maintenance-burden in the long-run.